posts/2024-03-11-crates-io-download-changes.md

```diff
@@ -0,0 +1,43 @@
+---
+layout: post
+title: "crates.io: Download changes"
+author: Tobias Bieniek
+team: the crates.io team <https://www.rust-lang.org/governance/teams/crates-io>
+---
+
+[crates.io](https://crates.io) is the official package registry of the Rust community where developers can publish their Rust packages for others to use. It plays a crucial role in the Rust ecosystem by providing a centralized platform for package distribution.
+
+The Rust community has been growing rapidly, and with it, the number of downloads from crates.io has been increasing by 200% to 300% per year.
+
+## The Problem
+
+This growth has brought with it some challenges. The most significant of these is that all download requests currently go through the crates.io API, occasionally causing scaling issues. If the API is down or slow, it affects all download requests too. In fact, the number one cause of waking up our crates.io on-call team is "slow downloads" due to the API having performance issues.
+
+Additionally, this setup is also problematic for users outside the United States, where download requests are slow due to the distance to the crates.io API servers.
```

Comment:
:canada: We were close enough to burn down the White House once, and so help us...

```suggestion
Additionally, this setup is also problematic for users outside of North America, where download requests are slow due to the distance to the crates.io API servers.
```